src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -5982,6 +5982,589 @@ public static class Extensions
                 Diagnostic(ErrorCode.ERR_EscapeVariable, "local2").WithArguments("local2").WithLocation(11, 18));
         }
 
+        [Fact]
+        public void Deconstruction_UnscopedRef()
```

Comment:
nit: it's a bit difficult to check that every scenario with `Deconstruct` invocation has an equivalent deconstruction-assignment scenario. Consider merging those pairs. 
For example, in `Deconstruction_UnscopedRef` this would mean adding:
```
                    R M2()
                    {
                        (var x, _) = new S();
                        return x;
                    }
```

Reply:
Thanks for the change. It's much easier to follow now :-)